Core.DomainServices/Queries/DataProcessingRegistrations/QueryReadModelsWhichShouldExpire.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Linq;
+using Core.DomainModel.GDPR.Read;
+
+namespace Core.DomainServices.Queries.DataProcessingRegistrations
+{
+    public class QueryReadModelsWhichShouldExpire : IDomainQuery<DataProcessingRegistrationReadModel>
+    {
+        private readonly DateTime _currentTime;
+
+        public QueryReadModelsWhichShouldExpire(DateTime currentTime)
+        {
+            _currentTime = currentTime;
+        }
+
+        public IQueryable<DataProcessingRegistrationReadModel> Apply(IQueryable<DataProcessingRegistrationReadModel> source)
+        {
+            var currentTime = _currentTime.Date;
+            return source.Where(
+                x =>
```

Comment:
this query is missing a prefilter on read models which are active (see the others)

As it is right now it evaluates all readmodels - even the ones that are already deactivated...

Notice the difference here:
![image](https://user-images.githubusercontent.com/13116889/208607270-32733479-3e9e-43f0-bc10-d050223757c6.png)